source/fem-dem/gls_vans.cc

```diff
@@ -558,13 +560,34 @@ GLSVANSSolver<dim>::quadrature_centered_sphere_method(bool load_balance_step)
               //***********************************************************************
               for (unsigned int n = 0; n < active_neighbors.size(); n++)
                 {
+                  double reference_sphere_volume;
```

Comment:
Why is the reference sphere volume calculated this deeply into the loops? Can't you calculate it at the beggining of the assembler instead? This does not depend on the neighbhour and on the particle (or maybe the particle, but not the neighbhour. I would pre-calculate everything I can to more the cost away.

Reply:
The reference sphere volume, depending on the parameter we choose might be defined using the cell size, the diameter of the particle, or cell volume. Therefore, I have to calculate it after the loop for the cells and the loop over particles, since it changes with these. It is determined twice, the first time when looping over neighbouring cells and neighbouring particles to determine the particle contribution and the second time to calculate the void fraction. I tried defining it before, but there are unknowns that are only known once you get inside these loops